test/remote-httpserverconnection.cpp

```diff
@@ -0,0 +1,557 @@
+	request.set(http::field::connection, "close");
+	request.content_length(0);
+	http::write(*client, request);
+	client->flush();
+
+	flat_buffer buf;
+	http::response<http::string_body> response;
+	BOOST_REQUIRE_NO_THROW(http::read(*client, buf, response));
+
+	BOOST_REQUIRE_EQUAL(response.version(), 11);
+	BOOST_REQUIRE_EQUAL(response.result(), http::status::unauthorized);
+	Dictionary::Ptr body = JsonDecode(response.body());
+	BOOST_REQUIRE(body);
+	BOOST_REQUIRE_EQUAL(body->Get("error"), 401);
+
+	BOOST_REQUIRE(Shutdown(client));
+}
+
+BOOST_AUTO_TEST_CASE(reuse_connection)
+{
+	CreateTestUsers();
+	SetupHttpServerConnection(true);
+
+	http::request<boost::beast::http::string_body> request;
+	request.method(http::verb::get);
+	request.target("/v1/test");
+	request.set(http::field::host, "localhost:5665");
+	request.set(http::field::accept, "application/json");
+	request.keep_alive(true);
+	http::write(*client, request);
+	client->flush();
+
+	flat_buffer buf;
+	http::response<http::string_body> response;
+	BOOST_REQUIRE_NO_THROW(http::read(*client, buf, response));
+
+	BOOST_REQUIRE_EQUAL(response.version(), 11);
+	BOOST_REQUIRE_EQUAL(response.result(), http::status::ok);
+	BOOST_REQUIRE_EQUAL(response.body(), "test");
+
+	request.keep_alive(false);
+	http::write(*client, request);
+	client->flush();
+
+	boost::system::error_code ec;
+	http::response_parser<http::string_body> parser;
+	BOOST_REQUIRE_NO_THROW(http::read(*client, buf, parser));
+
+	BOOST_REQUIRE(parser.is_header_done());
+	BOOST_REQUIRE(parser.is_done());
+	BOOST_REQUIRE_EQUAL(parser.get().version(), 11);
+	BOOST_REQUIRE_EQUAL(parser.get().result(), http::status::ok);
+	BOOST_REQUIRE_EQUAL(parser.get().body(), "test");
+
+	// Second read to get the end of stream error;
+	http::read(*client, buf, response, ec);
+	BOOST_REQUIRE_EQUAL(ec, boost::system::error_code{boost::beast::http::error::end_of_stream});
+
+	BOOST_REQUIRE(AssertServerDisconnected(std::chrono::seconds(5)));
+	BOOST_REQUIRE(Shutdown(client));
+	BOOST_REQUIRE(ExpectLogPattern("HTTP client disconnected .*", std::chrono::seconds(5)));
+}
+
+BOOST_AUTO_TEST_CASE(wg_abort)
+{
+	CreateTestUsers();
+	SetupHttpServerConnection(true);
+
+	UnitTestHandler::RegisterTestFn("wgjoin",
+		[this](HttpResponse&, const boost::asio::yield_context&) { m_WaitGroup->Join(); });
+
+	http::request<boost::beast::http::string_body> request;
+	request.method(http::verb::get);
+	request.target("/v1/test/wgjoin");
+	request.set(http::field::host, "localhost:5665");
+	request.set(http::field::accept, "application/json");
+	request.keep_alive(true);
+	http::write(*client, request);
+	client->flush();
+
+	flat_buffer buf;
+	http::response_parser<http::string_body> parser;
+	BOOST_REQUIRE_NO_THROW(http::read(*client, buf, parser));
+
+	BOOST_REQUIRE(parser.is_header_done());
+	BOOST_REQUIRE(parser.is_done());
+	BOOST_REQUIRE_EQUAL(parser.get().version(), 11);
+	BOOST_REQUIRE_EQUAL(parser.get().result(), http::status::ok);
```

Comment:
I'd also test that a body is also passed correctly. This should make the test more robust in the sense that the version and status are default values, so unless they are set to something else explicitly, these checks will always pass.